src/include/OpenImageIO/thread.h

```diff
@@ -725,8 +728,10 @@ class OIIO_API thread_pool {
     void deregister_worker(std::thread::id id);
     /// Is the thread in the pool or currently engaged in taking tasks from
     /// the pool?
+    bool is_worker(std::thread::id id) const;
```

Comment:
Do you need both the `const` and non-`const` versions?

Reply:
NVM, I guess this is purely for backwards compatibility?

Reply:
Yes, because it fixes a potential deadlock, I want to immediately backport it to the release branch, where I can't break link compatibility. I'm marking it as deprecated, so I'll remove the non-const next time we have a fully compatibility-breaking release.